content/blog/2019/07/2019-07-04-use-casc-for-exws.adoc

```diff
@@ -0,0 +1,141 @@
+---
+layout: post
+title: "Use Terraform with External Workspace Manager to Support EFS"
+tags:
+- jenkins
+- gsoc
+author: yufei_zhang
+description: "A brief example of using Terraform with EFS" # optional
+opengraph:
+---
+
+=== Why you want to use External Workspace Manager with EFS
+
+What is Terraform : A CasC software, like CloudFormation and Ansible designed for managing cloud resources. It can free you from managing cloud resource by console, and can be version controlled using git as it is configuration-as-code. Currently it supports AWS and Azure best. AWS CloudFormation is for AWS only.
+What is extenal-workspace-manager : It is a pipeline plugin which customizes the workspace path. The same disk can be mounted to multiple nodes to share resources
+What is EFS : Amazon Elastic File System, an Network File System, can be shared among a lot of EC2 instances.
+
+Who should read this blog :
+
+. build Jenkins based on cloud, want to have good version control over all resources including infrastructure; 
+. You want to share workspace among agents (slaves), there might be an upstream job creating a large file and don't want to copy it into downstream job's workspace because it's time-consuming; 
+. You're afraid of running out of workspaces, and wants an extensible disk.
+
+Who should not read this blog : 
+
+. Skilled AWS architecture already using CasC tools
+. Ops already using CasC and know how to provision EFS to a Jenkins node using user data.
+. Running Jenkins in local physical machine, and do not build Jenkins based on Clouds.
+
+What you'll want to do, is to learn external-workspace-manager directly to do similiar things, see blog link : link:https://github.com/jenkinsci/external-workspace-manager-plugin[External Workspace Manager]
+
+This tutorial use AWS as example, but other clouds like Google Cloud, Microsoft Azure are also supported. (same mechanism, different syntax), and here we go.
```

Comment:
`This tutorial use AWS as example` should read `This tutorial uses AWS as example`